components/forward_post_modal/forward_post_modal.tsx

```diff
@@ -0,0 +1,384 @@
+            <NotificationBox
+                variant={'info'}
+                text={notificationText}
+            />
+        );
+    }
+
+    const handleSubmit = useCallback(async () => {
+        let result = await actions.forwardPost(
+            post,
+            isPrivateConversation ? currentChannel.id : selectedChannelId,
+            comment,
+        );
+        if (result.error) {
+            setPostError(result.error);
+            return;
+        }
+
+        if (
+            selectedChannel?.details.type === Constants.MENTION_MORE_CHANNELS &&
+            selectedChannel?.details.type === Constants.OPEN_CHANNEL
+        ) {
+            result = await actions.joinChannelById(selectedChannelId);
+
+            if (result.error) {
+                setPostError(result.error);
+                return;
+            }
+        }
+
+        // only switch channels when we are not in a private conversation
+        if (!isPrivateConversation && selectedChannel) {
+            result = await actions.switchToChannel(selectedChannel.details);
+
+            if (result.error) {
+                setPostError(result.error);
+                return;
+            }
+        }
+
+        onHide();
+    }, [
+        actions,
+        post,
+        isPrivateConversation,
+        currentChannel.id,
+        selectedChannelId,
+        comment,
+        selectedChannel,
+        onHide,
+    ]);
+
+    const postPreviewFooterMessage = formatMessage({
+        id: 'forward_post_modal.preview.footer_message',
+        defaultMessage: 'Originally posted in ~{channelName}',
+    },
+    {
+        channel: currentChannel.display_name,
+    });
+
+    return (
+        <GenericModal
+            className='a11y__modal forward-post'
+            show={true}
+            enforceFocus={false}
+            autoCloseOnConfirmButton={false}
+            useCompassDesign={true}
+            modalHeaderText={formatMessage({
+                id: 'forward_post_modal.title',
+                defaultMessage: 'Forward Message',
+            })}
+            confirmButtonText={formatMessage({
+                id: 'forward_post_modal.button.forward',
+                defaultMessage: 'Forward',
+            })}
+            cancelButtonText={formatMessage({
+                id: 'forward_post_modal.button.cancel',
+                defaultMessage: 'Cancel',
+            })}
+            isConfirmDisabled={!canForwardPost}
+            handleConfirm={handleSubmit}
+            handleEnterKeyPress={handleSubmit}
+            handleCancel={onHide}
+            onExited={onHide}
+        >
+            <div className={'forward-post__body'}>
+                {isPrivateConversation ? (
+                    notification
+                ) : (
+                    <ForwardPostChannelSelect
+                        onSelect={handleChannelSelect}
+                        value={selectedChannel}
+                    />
+                )}
+                <Textbox
```

Comment:
This component is pretty big and it's doing a lot, can we further abstract it by moving some stuff into smaller components? Moving Textbox and it's functions might help